### PR TITLE
chore (refs T33659): add missing Hooks/Selectors.

### PR DESCRIPTION
**Pull request by ahmad-demos**
https://yaits.demos-deutschland.de/T33659

- add missing Hooks/Selectors.
- add dataCy attr to button UI Component.

### CODE REVIEW

File: templates/bundles/DemosPlanCoreBundle/DemosPlanUser/password_recover.html.twig [hunk 1]
```diff
@@ -19,12 +19,14 @@
                     control: { name: 'email' },
                     id: 'email',
                     type: 'email',
+                    dataCy: 'passwordRecoverForm:email',
```
Comment (spiess-demos) [inline templates/bundles/DemosPlanCoreBundle/DemosPlanUser/password_recover.html.twig:22]: I'd suggest to pass the attribute via `attributes` within the `control` field. See templates/bundles/DemosPlanCoreBundle/DemosPlanFaq/faq_admin_category_edit.html.twig:21 for reference.

File: templates/bundles/DemosPlanCoreBundle/UI/button.html.twig [hunk 2]
```diff
@@ -80,6 +80,10 @@
         disabled
     {% endif %}
 
+    {% if dataCy is defined %}
```
Comment (spiess-demos) [inline templates/bundles/DemosPlanCoreBundle/UI/button.html.twig:83]: If we're using `attributes`, we do not need the extra definition here.

**Review by spiess-demos — CHANGES_REQUESTED** [review]

File: templates/bundles/DemosPlanCoreBundle/DemosPlanUser/password_recover.html.twig (hunk 1 above)
Comment (ahmad-demos) [reply templates/bundles/DemosPlanCoreBundle/DemosPlanUser/password_recover.html.twig]: Done in https://github.com/demos-europe/demosplan-core/pull/1742/commits/60c5c909b95dfc5f604b18f71bbec1c4c4a6755b
Thanks!

File: templates/bundles/DemosPlanCoreBundle/UI/button.html.twig (hunk 2 above)
Comment (ahmad-demos) [reply templates/bundles/DemosPlanCoreBundle/UI/button.html.twig]: Done in https://github.com/demos-europe/demosplan-core/pull/1742/commits/60c5c909b95dfc5f604b18f71bbec1c4c4a6755b
Thanks!

**Review by spiess-demos — APPROVED** [review]
lgtm!

**Pull request merged**